databricks/resource_databricks_cluster_policy_test.go

```diff
@@ -0,0 +1,105 @@
+package databricks
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+
+	"github.com/databrickslabs/databricks-terraform/client/model"
+	"github.com/databrickslabs/databricks-terraform/client/service"
+)
+
+func TestAccClusterPolicyResourceFullLifecycle(t *testing.T) {
+	var policy model.ClusterPolicy
+	randomName := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	resource.Test(t, resource.TestCase{
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				// create a resource
+				Config: testExternalMetastore(randomName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccIDCallback(t, "databricks_cluster_policy.external_metastore",
+						func(client *service.DBApiClient, id string) error {
+							resp, err := client.ClusterPolicies().Get(id)
+							if err != nil {
+								return err
+							}
+							policy = *resp
+							return nil
+						}),
+					func(s *terraform.State) error {
+						if policy.Definition == "" {
+							return fmt.Errorf("Empty policy definition found")
+						}
+						return nil
+					},
+					resource.TestCheckResourceAttr("databricks_cluster_policy.external_metastore",
+						"name", fmt.Sprintf("Terraform policy %s", randomName)),
+				),
+			},
+			{
+				// add add the name for it
+				Config: testExternalMetastore(randomName + ": UPDATED"),
+				Check: resource.TestCheckResourceAttr("databricks_cluster_policy.external_metastore",
+					"name", fmt.Sprintf("Terraform policy %s", randomName+": UPDATED")),
+			},
+			{
+				Config:  testExternalMetastore(randomName + ": UPDATED"),
+				Destroy: true,
+				Check: testAccIDCallback(t, "databricks_cluster_policy.external_metastore",
+					func(client *service.DBApiClient, id string) error {
+						resp, err := client.ClusterPolicies().Get(id)
+						if err == nil {
+							return fmt.Errorf("Resource must have been deleted but: %v", resp)
+						}
+						return nil
+					}),
+			},
```

Comment:
additional test case for handling deleted policy and policy recreation

Reply:
done